log/log.go

```diff
@@ -33,6 +34,8 @@ func Setup(config Config) {
 
 // Build is a panic-free version of Setup.
 func Build(config Config) error {
+	logMutex.Lock()
+	defer logMutex.Unlock()
```

Comment:
`Build` is called only from `main` before the the application starts. There is no need for locking.

In other cases where on-demand singleton initialization is required, [Once.Do](https://golang.org/src/sync/once.go) or [Once.Try](https://github.com/go-ocf/kit/blob/master/sync/once.go) can be used.

Reply:
Yes and no. I need it for tests because i get concurrency access to the memory with `go test -race` . And I for that I want to leave it as is.

Reply:
I disagree with paying this penalty for every log message for the sake of testing.

Reply:
i want to use test with race arguments - i can change it to atomic.Value but without any sync prim i cannot able to remove concurrency issue. It seems that we need to define something like logger which will be constructed in main and passed to services.

Reply:
[Once.Do](https://golang.org/src/sync/once.go) would do the trick